### PR TITLE
Use os.CreateTemp in kubectl editor

### PR DESCRIPTION
**Pull request by xuhdev**
`os.CreateTemp` seems to perform the exactly same task here, and its
implementation seems having considered many more edge cases than the
implementation here. This patch uses `os.CreateTemp` here to avoid
reinventing the wheel.

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

See the top of this comment box.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by dougsland — COMMENTED** [review]
To me make sense but lets hear from others members.
/cc @eddiezane 

/priority backlog
/ok-to-test
/lgtm

**Comment by eddiezane** [comment]
`ioutil` [is deprecated](https://golang.org/doc/go1.16#ioutil) as of Go 1.16.

Can you please swap this to [`os.CreateTemp`](https://golang.org/pkg/os/#CreateTemp)?

It looks like this code was [imported from OpenShift back in 2015](https://github.com/kubernetes/kubernetes/pull/13604) and there was a [reason](https://github.com/kubernetes/kubernetes/pull/13604/files#r38999530) for this implementation. Can you confirm that this is resolved?

/cc @smarterclayton

**Review by dougsland — COMMENTED** [review]
/retest

**Comment by rikatz** [comment]
Hey @xuhdev thanks for the PR.

Overall this seems a good cleanup to me.

Will wait for some more feedback (and anyway we need to wait not until the end of code freeze).

Thanks!

/test pull-kubernetes-e2e-kind

**Comment by xuhdev** [comment]
The test failure doesn't seem to be relevant. I rebased the PR so we can see if things change.

**Comment by rikatz** [comment]
/lgtm

I guess this is a good addition at all :)

File: staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go:163]: Looks like this wrapping function isn't used elsewhere so let's remove it for extra cleanup!

**Review by eddiezane — COMMENTED** [review]
One more thing then lgtm!

File: staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go
Comment (xuhdev) [reply staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go]: Definitely! Done

**Comment by eddiezane** [comment]
Please squash commits.

**Comment by rikatz** [comment]
@eddiezane I've learned this trick and try to tell everyone about it:

/label tide/merge-method-squash

And now Tide will squash the commits before merging :)

**Comment by eddiezane** [comment]
@rikatz oooo magic!

@xuhdev `-26` is a great first contribution 😄 

/approve

**Comment by eddiezane** [comment]
@xuhdev that commit message seems to be for something else?

**Comment by xuhdev** [comment]
Oops, I copied the wrong commit message when I was squashing 😅 Corrected now.

**Comment by eddiezane** [comment]
Failure looks like https://github.com/kubernetes/kubernetes/issues/99801

/test pull-kubernetes-e2e-kind

**Comment by rikatz** [comment]
/lgtm
Thank you @xuhdev !

**Review by dougsland — COMMENTED** [review]
/lgtm

**Comment by xuhdev** [comment]
Does this PR have to be milestoned?

/milestone v1.21

**Comment by rikatz** [comment]
We're on code freeze so this will need to wait until v1.21 is released to be merged :)

**Review by rikatz — APPROVED** [review]

**Comment by xuhdev** [comment]
Thanks! Could someone change the milestone to v1.21? The tide bot says it won't be mergeable.

**Comment by rikatz** [comment]
We're on code freeze so this will need to wait until v1.21 is released to be merged :) 

Once v1.21 is released, this restriction will be removed :)

**Pull request merged**
